magma/backend/mlir/translation_unit.py

```diff
@@ -108,6 +121,15 @@ def compile(self):
                 hardware_module.compile()
                 if hardware_module.hw_module:
                     self.set_hardware_module(dep, hardware_module)
+        if self._opts.split_verilog:
+            if self._opts.basename is None or self._opts.suffix is None:
+                raise ValueError(
+                    "Must specify basename and suffix if split_verilog is set"
+                )
+            _prepare_for_split_verilog(
+                self._hardware_modules.values(),
+                f"{self._opts.basename}.{self._opts.suffix}",
```

Comment:
Since we're just using a specific target file here, is this any different than the current verilog export? Does that support a directory structure in the name? Would simply omitting the basename and just providing a directory here achieve the eventual result desired for bind?

Reply:
Good question. The way `--export-verilog`/`--export-split-verilog` work is as follows:

- If `--export-verilog` is specified, all `hw.output_file` attributes are ignored and verilog is output to stdout, but the output verilog is annotated with a comment that separates the various specified output files.
- If `--export-split-verilog` is specified, any module annotated with the `hw.output_file` attribute is written out to the specified file, and any un-annotated module is written out to `<module name>.sv` in the CWD. If `hw.output_file` is a directory, then it is the same as being un-annotated, in that the verilog is written out to `<dir>/<module name>.sv`.

In our case, there will be *no un-annotated modules if `split_verilog` is specified. Currently, the only annotation is for `<basename>.v` (or `.sv`). However, this PR is in preparation for separating monitor files, which requires specifying all modules.